crypto/sha/asm/sha512-x86_64.pl

```diff
@@ -2182,6 +2236,8 @@ ()
 	add	$a1,$A
 	#mov	`2*$SZ*$rounds+8`(%rsp),$inp	# $_inp
 	lea	`2*$SZ*($rounds-8)`(%rsp),%rsp
+# restore frame pointer to original location at $_rsp
+.cfi_cfa_expression	$_rsp,deref,+8
```

Comment:
This is again common between sha256 and sha512, after adjusting the %rsp
the secondary frame pointer is gone, and need to switch back to the setting before the loop.